### ISSUE TITLE
2020-11-4: When building the Jekyll site locally, how can I get pages other than the main one to work?

### ISSUE THREAD

**Issue by baileythegreen** [carpentries/glosario#205]
Note: this is a problem I have seen with all Jekyll sites I have worked with (not a large number, but bigger than one), so it may be that I have done something wrong.

I was hoping to make and try out some changes, but the changes I wanted to try are at the individual language level, and those pages render on GitHub, but not locally. Whenever I google this issue, I can only find threads where people had the opposite problem, so I really don't know where to begin.

Any ideas?

**Comment by baileythegreen** [comment]
Update:

I got this to work by altering `index.html` like this:

```
{% for lang in site.languages %}
  <!--
  {% assign link = lang.key | prepend: '/' | append: '/' | relative_url %}
  <h2><a href="{{link}}">{{lang.name}}</a></h2>
  -->                                                                        # comment these lines out
  <h2><a href="{{site.baseurl}}/{{lang.key}}">{{lang.name}}</a></h2>         # add this line
  <div dir='auto'>
  {{lang.blurb | markdownify}}
  </div>
{% endfor %}
```

I'm not entirely clear on why I need to do this for it to work, or whether this is a problem for others, but if it is, a change like this will help people wanting to test changes to the web version of Glosario.

**Comment by baileythegreen** [comment]
I forgot about one other change: I had to create a `_data/` directory and place `glossary.yml` there for the individual language pages to show any content.

This, so the `{% assign gloss = site.data.glossary %}` line in `_incudes/glossary.html` points to something.

**Comment by BatoolMM** [comment]
Hi @baileythegreen, I had the same problem initially because I was using `bundle exec jekyll s -l -o` rather than `make serve` when building the site locally. This is because `make serve` copies `./glossary.yml` into `./_data/glossary.yml` and then runs `bundle exec jekyll`. I was confused at first until I went through all the closed PRs and found this PR #19.

**Comment by baileythegreen** [comment]
@BatoolMM Thanks! I see now that the gh_pages branch has the _data folder, which explains why it works fine, but my original attempt at forking and rendering my own version failed. I hadn't managed to find PR #19, probably because the subject line is not directly related to the terms I searched on.

I've been using `bundle exec jekyll serve --livereload` because somewhere I found that when I first did something with GitHub Pages and because I don't know enough to know any different. Yet. I'll think about what to do in my local versions, but for now have put the files affected by my workaround in `.gitignore` so I want accidentally end up pushing changes others don't need.
